src/sinks/util/socket_bytes_sink.rs

```diff
@@ -50,22 +50,21 @@ where
     pub(crate) fn new(
         inner: T,
         shutdown_check: impl Fn(&mut T) -> ShutdownCheck + Send + 'static,
-        acker: Acker,
         socket_mode: SocketMode,
     ) -> Self {
         Self {
             inner: FramedWrite::new(inner, BytesCodec::new()),
             shutdown_check: Box::new(shutdown_check),
             events_total: 0,
             bytes_total: 0,
-            acker,
             socket_mode,
+            finalizers: Vec::new(),
         }
     }
 
     fn ack(&mut self) {
         if self.events_total > 0 {
-            self.acker.ack(self.events_total);
+            drop(std::mem::take(&mut self.finalizers));
```

Comment:
Where are we actually populating `self.finalizers`?

Reply:
Hmm, we aren't. I must be missing a step here.

Reply:
Done in 37861a004